server/mqtt.go

```diff
@@ -4618,29 +4618,30 @@ func mqttIsRetained(flags byte) bool {
 	return flags&mqttPubFlagRetain != 0
 }
 
-// Checks if the subject is in the form of a Sparkplug BIRTH or DEATH message,
-// MQTT topic like spBv1.0/{group_id}/{message_type}/{edge_node_id}/{device_id}
-func sparkbIsBirth(natsSubject, prefix []byte) (is bool, typ, groupID, nodeID, deviceID string) {
-	if !bytes.HasPrefix(natsSubject, prefix) {
-		return false, _EMPTY_, _EMPTY_, _EMPTY_, _EMPTY_
-	}
-	natsSubject = natsSubject[len(prefix):]
-	parts := bytes.Split(natsSubject, []byte("."))
-	if len(parts) < 3 {
-		return false, _EMPTY_, _EMPTY_, _EMPTY_, _EMPTY_
+func sparkbParseBirthDeathTopic(topic []byte) (isBirth, isDeath, isCertificate bool) {
+	if bytes.HasPrefix(topic, sparkbCertificatesTopicPrefix) {
+		isCertificate = true
+		topic = topic[len(sparkbCertificatesTopicPrefix):]
 	}
-	typ = string(parts[1])
-	isDevice := (typ == sparkbDBIRTH)
-	if typ != sparkbNBIRTH && !isDevice {
-		return false, _EMPTY_, _EMPTY_, _EMPTY_, _EMPTY_
+	if !bytes.HasPrefix(topic, sparkbNamespaceTopicPrefix) {
+		return false, false, false
 	}
+	topic = topic[len(sparkbNamespaceTopicPrefix):]
 
-	groupID = string(natsSubjectToMQTTTopic(parts[0]))
-	nodeID = string(natsSubjectToMQTTTopic(parts[2]))
-	if isDevice && len(parts) > 3 {
-		deviceID = string(natsSubjectToMQTTTopic(parts[3]))
+	parts := bytes.Split(topic, []byte("/"))
+	if len(parts) < 3 || len(parts) > 4 {
+		return false, false, false
+	}
+	typ := string(parts[1])
```

Comment:
Using `typ := bytesToString(parts[1])` here I think would be safe since it is a conversion just for checking content but we don't reference `typ` outside of here.